### ISSUE TITLE
Serialization method on base class?

### ISSUE THREAD

**Issue by 5ebastianMeier** [essential-projects/errors_ts#6]
#### Describe your issue

I currently have the problem that errors are serialized when they travel through the messagebus and that is currently done by just extracting the `message`-property.

This makes the error unidentifiable at a later point. To work around this issue we would need a proper error serialization.

#### How can others reproduce the issue?

-

#### Possible solution

I would like to suggest a fallback serialization method on the error base class that can be overwritten by deriving error classes. That way we can encapsulate the serialization in the error class and use it via a common method that is present on all of our errors.

The above could look like this:
```
  public toJSON(): ISerializedError {

    return {
      code: this.code,
      info: this.message,
    };
  }
```

#### What's your setup

-

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

**Comment by 5ebastianMeier** [comment]
I just noticed another issue:

When we serialized the error as described above we lose the type information - so we would additionally need something like this in the base class:

```
  constructor() {
    this.name = this.constructor.name;
  }
```

**Comment by heikomat** [comment]
How about some more info?
```
  public toJSON(): ISerializedError {
    return {
      errorClassName: this.constructor.name,
      code: this.code,
      message: this.message,
      callStack: this.stack
    };
  }
```

especially with we could at least partly deserialize the error:
```
toError(serializedError: string) {
  const errorInfo = JSON.parse(serializedError);
  const errorClass = new errorClasses[errorInfo.errorClassName](errorInfo.code, errorInfo.message);
  errorClass.stack = errorInfo.callStack;
  return errorClass
}
```

**Comment by heikomat** [comment]
This was implemented in #7
